File: cs2go/TestRole.go
package main

type TestRole struct {
	Attack  int
	d       map[string]int
	Defense int
	Flg     bool
	Strs    []int
}

var Name = "MyName"

func (tn *TestRole) GetHurt(testRole TestRole, callNum int) int {
	tn.Show()
	StShow()
	var num = 100
	var role = new(TestRole)
	var dd = num*100 + 100
	var num3 = num + 1
	var len = cap(role.Strs)
	var num1 = 101
	for i := 10; i > 100; i++ {
		num1++
	}
	return cap(testRole.Strs)
}
func (tn *TestRole) TestFunc(num int, name string) int {
	return 100
}
func ShowTest(num int) int {
	return 100
}
func (tn *TestRole) Show() {
}
func StShow() {
}

File: cs2go/TestRole.cs
using System.Collections.Generic;



[Test]
public class TestRole:ITestInterface
{
    public static string Name = "MyName";
    public int Attack;

    public Dictionary<string, int> d;
    public int Defense;

    public bool Flg;

    public int[] Strs;


    public int GetHurt(TestRole testRole, int callNum)
    {
        Show();
        StShow();
        int num = 100;
        TestRole role = new TestRole();
        int dd = num * 100 + 100;
        int num3 = num + 1;

        int len = role.Strs.Length;
        int num1 = 101;
        for (var i = 10; i < 100; i++)
        {
            num1++;
        }


        return testRole.Strs.Length;
    }


    public int TestFunc(int num, string name)
    {
        return 100;
    }


    public static int ShowTest(int num)
    {
        return 100;
    }

    public void Show()
    {
    }

    public static void StShow()
    {
    }

    public int GetMaxAttack()
    {
        return 100;
    }
}